core/src/mast/node/basic_block_node/arbitrary.rs

```diff
@@ -207,31 +228,218 @@ impl Arbitrary for MastForest {
+                                    .map(|i| i % num_basic_blocks)
+                                    .collect::<Vec<_>>()
+                            },
+                        ),
+                        // Generate indices for call nodes (need 1 child each)
+                        prop::collection::vec(any::<usize>(), num_calls..=num_calls).prop_map(
+                            move |indices| {
+                                indices
+                                    .into_iter()
+                                    .map(|i| i % num_basic_blocks)
+                                    .collect::<Vec<_>>()
+                            },
+                        ),
+                        // Generate digests for external nodes
+                        prop::collection::vec(any::<[u64; 4]>(), num_externals..=num_externals)
+                            .prop_map(move |digests| {
+                                digests
+                                    .into_iter()
+                                    .map(|[a, b, c, d]| {
+                                        Word::from([
+                                            Felt::new(a),
+                                            Felt::new(b),
+                                            Felt::new(c),
+                                            Felt::new(d),
+                                        ])
+                                    })
+                                    .collect::<Vec<_>>()
+                            }),
+                    )
+                },
+            )
+            .prop_map(
+                move |(
+                    basic_blocks,
+                    decorators,
+                    (_num_joins, _num_splits, _num_loops, _num_calls, _num_externals, num_dyns),
+                    join_pairs,
+                    split_pairs,
+                    loop_indices,
+                    call_indices,
+                    external_digests,
+                )| {
+                    let mut forest = MastForest::new();
+
+                    // 1) Add all decorators first
+                    for decorator in decorators {
+                        forest.add_decorator(decorator).expect("Failed to add decorator");
+                    }
+
+                    // 2) Add basic blocks and collect their IDs
+                    let mut basic_block_ids = Vec::new();
+                    for block in basic_blocks {
+                        let node_id = forest.add_node(block).expect("Failed to add basic block");
+                        basic_block_ids.push(node_id);
+                    }
+
+                    // 3) Add control flow nodes in topological order (children already exist)
+                    let mut all_node_ids = basic_block_ids.clone();
+
+                    // Add join nodes
+                    for &(left_idx, right_idx) in &join_pairs {
+                        if left_idx < all_node_ids.len() && right_idx < all_node_ids.len() {
```

Comment:
We expect the children to exist, right? Hence, shouldn't we `assert!` that this condition is true to catch bugs, instead of "silently failing"?

Comment applies everywhere else we do this (e.g. the split nodes below, etc).